DependencyInjection/Compiler/DoctrineDBALLoggerPass.php

```diff
@@ -0,0 +1,146 @@
+
+        if (!$reflectionClass) {
+            throw new InvalidArgumentException(sprintf(
+                'Class "%s" used for service "%s" cannot be found.',
+                $serviceClass,
+                $serviceId
+            ));
+        }
+
+        if (!$reflectionClass->implementsInterface(SQLLogger::class)) {
+            throw new InvalidArgumentException(sprintf(
+                'The service "%s" tagged "%s" must implement "%s".',
+                $serviceId,
+                self::TAG_NAME,
+                SQLLogger::class
+            ));
+        }
+
+        if ($reflectionClass->isAbstract()) {
+            throw new InvalidArgumentException(sprintf(
+                'The service "%s" tagged "%s" cannot be abstract.',
+                $serviceId,
+                self::TAG_NAME
+            ));
+        }
+    }
+
+    private function createChainDefinitionAttachedToConnection(
+        ContainerBuilder $container,
+        string $connectionName,
+        $configurationDefinition
+    ): Definition {
+        $configuredLoggerReference = $this->getSQLLoggerFromConnection($configurationDefinition);
+
+        if ($configuredLoggerReference === null) {
+            return $this->createChainAndAttachToConnection($container, $connectionName, $configurationDefinition);
+        }
+
+        $configuredLoggerDefinition = $container->getDefinition((string)$configuredLoggerReference);
+
+        if ($this->isChainLogger($configuredLoggerDefinition)) {
+            return $configuredLoggerDefinition;
+        }
+
+        $chainDefinition = $this->createChainAndAttachToConnection($container, $connectionName, $configurationDefinition);
+        $chainDefinition->addMethodCall('addLogger', [$configuredLoggerReference]);
+
+        return $chainDefinition;
+    }
+
+    private function getSQLLoggerFromConnection(Definition $configurationDefinition): ?Reference
+    {
+        $methodCallList = $configurationDefinition->getMethodCalls();
+
+        foreach ($methodCallList as $methodCall) {
+            if ($methodCall[0] !== 'setSQLLogger') {
+                continue;
+            }
+
+            return $methodCall[1][0];
+        }
+
+        return null;
+    }
+
+    private function isChainLogger(Definition $serviceDefinition): bool
+    {
+        $methodCallList = $serviceDefinition->getMethodCalls();
+
+        foreach ($methodCallList as $methodCall) {
+            if ($methodCall[0] === 'addLogger') {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function createChainAndAttachToConnection(
+        ContainerBuilder $container,
+        string $connectionName,
+        Definition $configurationDefinition
+    ): Definition {
+        $chainId = self::BASE_CHAIN_NAME . '.' . $connectionName;
+
+        $chainReference = new Reference($chainId);
+        $chainDefinition = new Definition(self::BASE_CHAIN_NAME);
```

Comment:
Shouldn't there be `LoggerChain::class`?

I am getting error `Symfony\Component\DependencyInjection\Exception\InvalidArgumentException: "'doctrine.dbal.logger.chain'" is not a valid class name for the "doctrine.dbal.logger.chain.default" service.` with self::BASE_CHAIN_NAME